### ISSUE TITLE
quick show can't auto hide

### ISSUE THREAD

**Issue by over140** [SwiftKickMobile/SwiftMessages#77]
my code :

`

    fileprivate lazy var toastConfig: SwiftMessages.Config = {
        var config = SwiftMessages.defaultConfig
        config.presentationStyle = .top
        config.shouldAutorotate = true
        config.interactiveHide = true
        return config
    }()

    override func viewDidLoad() {
        super.viewDidLoad()
        SwiftMessages.pauseBetweenMessages = 0
    }

    fileprivate func toast(text: String, backgroundColor: UIColor, timeInterval: TimeInterval = 2.5) {
        let toastView = makeToastView()
        toastView.configureTheme(backgroundColor: backgroundColor, foregroundColor: .white)
        toastView.configureContent(body: text)
        toastConfig.presentationContext = .window(windowLevel: UIWindowLevelStatusBar)
        toastConfig.duration = .seconds(seconds: timeInterval)
        SwiftMessages.hide()
        SwiftMessages.show(config: toastConfig, view: toastView)
    }

    fileprivate func makeToastView() -> MessageView {
        let view = MessageView.viewFromNib(layout: .CardView)
        view.configureTheme(backgroundColor: .minewaveLightRed, foregroundColor: .white)
        view.bodyLabel?.textAlignment = .center
        view.titleLabel?.isHidden = true
        view.button?.isHidden = true
        view.iconImageView?.isHidden = true
        view.iconLabel?.isHidden = true
        view.configureDropShadow()
        return view
    }

`

If I quick call toast func, like fast click button, the MessageView maybe can't auto hide.

**Comment by crossle** [comment]
Same problem, when invoke multiple hide and show on SwiftMessages, the MessageView can't hide

**Comment by wtmoose** [comment]
Make sure you're on 3.0.1. There was a bug in 3.0.0.

**Comment by wtmoose** [comment]
Here's a working test project with 3.0.1:

[AutoHideTest.zip](https://github.com/SwiftKickMobile/SwiftMessages/files/855027/AutoHideTest.zip)

**Comment by crossle** [comment]
Use lastest 3.3.1 version, your demo still have same problem, quick click

**Comment by over140** [comment]
Thinks your help!  Your sample still have the problem, I record it and upload to YouTube : https://youtu.be/ANjSwZJPEQs

When you fast click button Five times per second

**Comment by wtmoose** [comment]
Thanks for clarifying. I can reproduce that. I'll take a look.

**Comment by wtmoose** [comment]
This problem was introduced by pull request #63.

The problem relates to duplicate messages, i.e. messages that have the same `identifier`. Previously, SwiftMessages ignored duplicate messages until the current message *finished* hiding. The change made by #63 is to ignore duplicates until a message *begins* hiding. But the implementation is not robust.

My preferred solution is:
1. Revert #63. The timing of when to stop ignoring duplicates is an arbitrary decision and I prefer the original behavior.
2. Use `config.ignoreDuplicates = false` or set `message.customId = ...` if you want to display a duplicate message.

But I'm also open to fixing #63 or implementing #65, which I think is a better way to address the use case discussed in #63.

My question for you is, what behavior do you want and why?

Here is the sample project with #63 reverted.

[AutoHideTest2.zip](https://github.com/SwiftKickMobile/SwiftMessages/files/857539/AutoHideTest2.zip)

Thanks for your input.

cc @srstanic

**Comment by over140** [comment]
I used your Project AutoHideTest2.zip' and changed this code to add other button :

```

    @IBAction func test1(_ sender: Any) {
        toast(text: "Test1", backgroundColor: .darkGray)
    }

    @IBAction func test2(_ sender: Any) {
        toast(text: "Test2", backgroundColor: .blue)
    }

```

If I alternate click the buttons , the problem still.

Here is the sample project
[AutoHideTest3.zip](https://github.com/SwiftKickMobile/SwiftMessages/files/860440/AutoHideTest3.zip)

Here is the record video
https://youtu.be/ShUKS2O62Js

Thanks for your help.

**Comment by srstanic** [comment]
Hey guys, I'll take a look at it today and let you know my thoughts.

**Comment by srstanic** [comment]
@wtmoose can you please clarify which change from #63 you think introduced this issue? It should only affect whether a message is going to be enqueued or not. Looking from a video @over140 posted it's visible at the end of the 1nd second that a message fails to be hidden. I'll try to figure out why and let you know what I find.

**Comment by wtmoose** [comment]
@over140 Thanks, I'll take a look.

@srstanic Based on the followup comments from @over140, your change expanded an underlying bug to a wider set of scenarios. I don't think you need to look at anything.

**Comment by srstanic** [comment]
@wtmoose I wanted to help but as far as I can tell, it would require a wider design change to fix this. I've used the following code in the ```test()``` method of the ViewController to reproduce the bug:
```
for i in 1...3 {
    let delay = 100*i
    DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + DispatchTimeInterval.milliseconds(delay)) {
        self.toast(text: String(i), backgroundColor: .darkGray)
    }
}
```
And I found that message 2 always fails to hide, but the circumstances may differ depending on where and if you put a breakpoint.

The solution would really depend on what we expect to happen as a result of the ```SwiftMessages.hide()``` method. Does it affect only the message that is currently displayed, or if the current message is already hiding, should it then affect the next message in the queue? 

Anyway, let me know if I can be of any help further on.
